cameraview/src/main/java/com/otaliastudios/cameraview/picture/Full1PictureRecorder.java

```diff
@@ -67,7 +70,21 @@ public void onPictureTaken(byte[] data, final Camera camera) {
                         mResult.data = data;
                         mResult.rotation = exifRotation;
                         LOG.i("take(): starting preview again. ", Thread.currentThread());
+
                         camera.setPreviewCallbackWithBuffer(mEngine);
+                        Size previewStreamSize = mEngine.getPreviewStreamSize(Reference.SENSOR);
+                        if (previewStreamSize == null) {
+                            throw new IllegalStateException("Preview stream size " +
+                                    "should never be null here.");
+                        }
+                        // Need to re-setup the frame manager, otherwise no frames are processed
+                        // after takePicture() is called
+                        mEngine.getFrameManager().setUp(
+                                mEngine.getFrameProcessingFormat(),
+                                previewStreamSize,
+                                mEngine.getAngles()
+                        );
+
                         camera.startPreview(); // This is needed, read somewhere in the docs.
```

Comment:
Could you put all these lines (74-88) into a condition `if (mEngine.getState().isAtLeast(CameraState.PREVIEW))` ? This way we're sure that the preview is still running and previewStreamSize is really not null as you said.